### PR TITLE
docs: Update debug guidelines for Fault scenarios

### PR DESCRIPTION
**Pull request by kishore-sn**
Update debug guidelines for all fault scenarios in the HowToDebugACrash.md file.

Signed-off-by: Kishore S N <kishore.sn@samsung.com>

### CODE REVIEW

File: docs/HowToDebugACrash.md [hunk 1]
```diff
@@ -161,3 +162,110 @@ Callstack from values in stackdump
...
+
+Armv8-m : https://developer.arm.com/documentation/ddi0553/latest
+
+### MemManage Fault Status Register (MMFSR)
+
+1. ```MMARVALID```
+
+Debug msg: Access violation occurred at address (MMFAR) : 0xAddress
+
+This message indicates that the address which caused the memfault is available to us. Check the printed address value. If the address is zero, it is probably caused due to a null pointer access. If the address value is non-zero, then the fault can the caused due to one of the following reasons.
```
Comment (abhishek-samsung) [inline docs/HowToDebugACrash.md:179]: Please change "then the fault can 'the' caused due to" --> "then the fault can 'be' caused due to"

File: docs/HowToDebugACrash.md [hunk 2]
```diff
@@ -161,3 +162,110 @@ Callstack from values in stackdump
...
+Debug msg: Data access violation occurred.
+
+Use the PC value and the address printed in the debug message to find which variable is causing an invalid data access as mention in point (1) above.
+
+
+3. ```IACCVIOL```
+
+Debug msg: Instruction access violation occurred while fetching instruction from an Execute Never (XN) region.
+
+Execution from an MPU region with no execute permission. Check the line number of PC and make sure that corresponding MPU region has the proper previlages.
```
Comment (jeongchanKim) [inline docs/HowToDebugACrash.md:208]: `previlages` -> `privileges`

File: docs/HowToDebugACrash.md [hunk 3]
```diff
@@ -161,3 +162,110 @@ Callstack from values in stackdump
...
+In both the above cases, the developer can use the addr2line tool along with the PC and LR values to find the line number in code, where the error has occured. Please note the following points:
+- The LR value points to the next instruction to be executed after return from a function call. So, the LR will point to a line number after a function in which the fault would have occured.
+- The LR and PC might not contain exact line numbers if the function was declared as "inline", or if some compiler optimization has caused the function to be inlined.
+
+
+2. ```DACCVIOL```
+
+Debug msg: Data access violation occurred.
+
+Use the PC value and the address printed in the debug message to find which variable is causing an invalid data access as mention in point (1) above.
```
Comment (dineshdas-samsung) [inline docs/HowToDebugACrash.md:201]: mention - > mentioned

File: docs/HowToDebugACrash.md [hunk 4]
```diff
@@ -161,3 +162,110 @@ Callstack from values in stackdump
...
+
+The only difference lies in the IMPRECISERR bit. This bit is set if an imprecise data access violation has occured. It means that the current PC value does NOT point to the instruction with caused the data access violation. The violation is caused by some previous instruction, but was delayed because memory accesses are not always synchronous. In order to debug this type of fault, the developer can check the previous few lines of code starting from the current PC value. The developer can also use the objdump tool to print the disassembly of the code at the current PC value and then check each assembly instruction which is either loading or storing values into the memory.
```
Comment (dineshdas-samsung) [inline docs/HowToDebugACrash.md:231]: Did u mean  this -> instruction  "which "caused the data access violation?

File: docs/HowToDebugACrash.md [hunk 5]
```diff
@@ -161,3 +162,110 @@ Callstack from values in stackdump
...
+
+5. ```MLSPERR```
+
+Debug msg: Error occurred during lazy state preservation of Floating Point unit registers.
+
+This bit also indicates an error during stacking of registers during exception handling.
+
+
+### Bus Fault Status Register (BFSR)
+A bus fault is vary similar to memory management fault as mentioned above. Any access of data or instructions from an invalid memory address can cause a bus fault. If the MPU is enabled, then most of such invalid accesses will result in a corresponding memfault. However, if the MPU is disabled, all invalid memory accesses result in bus fault. Bus faults can either be data abort or prefetch abort, based on whether the fault occured while accessing data or instructions. The BFARVALID, STKERR, UNSTKERR, LSPERR fields of this register are similar to the corresponding bits in the memfault scenario. The IBUSERR bit indicates an instruction access fault; whereas, the PRECISERR and IMPRECISERR bits indicate a data access violation. The PRECISERR is similar to DACCVIOL in the memfault register.
```
Comment (jeongchanKim) [inline docs/HowToDebugACrash.md:229]: Is this intended to use `vary`, not `very`?

**Pull request merged**